dashboard/lib/projects_list.rb

```diff
@@ -116,6 +117,33 @@ def fetch_featured_published_projects
       return featured_published_projects
     end
 
+    # Retrieve a class set of libraries for a specified class section
+    # @param section The section that has all students whose libraries should
+    #                be returned.
+    # @return [Hash<Array<Hash>>] A hash of lists of published libraries.
+    def fetch_section_libraries(section)
+      project_types = PUBLISHED_PROJECT_TYPE_GROUPS[:library]
+      section_students = section.students
+      [].tap do |projects_list_data|
+        student_storage_ids = PEGASUS_DB[:user_storage_ids].
+          where(user_id: section_students.pluck(:id)).
+          select_hash(:id, :user_id)
+        student_storage_id_list = student_storage_ids.keys
+        PEGASUS_DB[:storage_apps].
+          where(storage_id: student_storage_id_list, state: 'active').
```

Comment:
Nice work cleaning up this query! Our database utilization graphs will thank you :-)